tests/fixture/test_data_server.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fixture
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+)
+
+const (
+	allocateKeysPath                  = "/allocateKeys"
+	keyCountParameterName             = "count"
+	requestedKeyCountExceedsAvailable = "requested key count exceeds available allocation"
+)
+
+var (
+	errRequestedKeyCountExceedsAvailable = errors.New(requestedKeyCountExceedsAvailable)
+	errInvalidKeyCount                   = errors.New("key count must be greater than zero")
+)
+
+type TestData struct {
+	FundedKeys []*secp256k1.PrivateKey
+}
+
+// http server allocating resources to tests potentially executing in parallel
+type testDataServer struct {
+	TestData
+
+	// Synchronizes access to test data
+	mutex sync.Mutex
```

Comment:
minor standardization style nit
```suggestion
	lock sync.Mutex
```